### DISCUSSION TITLE
Table element not droping

### DISCUSSION THREAD

**Discussion by gxanshu** [GrapesJS/grapesjs#5217] [Q&A] [answered]
recently i tried to create blocks for grapesjs to handle tables here is my code 
```javascript
const commonBlockProps = {
  category: "blocks",
  select: true
};

const blocks = [
  {
    ...commonBlockProps,
    label: "Box",
    content: `<table style="width:100%; min-height: 50px;"></table>`
  },
  {
    ...commonBlockProps,
    label: "Body",
    content: `<tbody data-gjs-type="tbody" data-gjs-droppable="true" style="height: 75px;"></tbody>`
  },
  {
    ...commonBlockProps,
    label: "Row",
    content: `<tr data-gjs-type="row" data-gjs-droppable="true" style="height: 75px;"></tr>`
  },
  {
    ...commonBlockProps,
    label: "Column",
    content: `<td data-gjs-draggable="true" data-gjs-type="cell" data-gjs-resizable="width" style="width: calc(100% / 7);"></td>`
  }
];

blocks.map((template) => {
      editor.BlockManager.add(template.id, template)
    })
```
i got successfully 3 blocks in block manager then i tried to drop Box block (table block) first then try to add tbody or tr or td nothing works.

nothing able to drop in it. everytime it warns `Invalid target position` in console. i know that the table component only accept tbody, thead and tfooter and the tr tag only accept td but when i try to add tr in tbody then it did'nt work also when i try to add td in tr then also not work

**Comment by artf** [comment, marked as answer]
Unfortunately this issue is due to the native `DOMParser` which outputs an empty document if you try to pass an invalid syntax. For example this what is passed in the case of `Column`:
```html
<body>
  <td ...></td>
</body>
```
as having only `<td>` without the full table around it is an invalid syntax, the result will be empty and the editor won't be able to know what component to create.

The workaround here is to avoid the parser and pass directly the component definition, eg:
```js
{
    ...commonBlockProps,
    label: "Column",
    content: { 
      type: 'cell', 
      // ...
      style: { width: 'calc(100% / 7)' } 
    },
  }
```

**Reply by gxanshu** [reply]
Thank you very much
